sparkles/roll_optimize.py

```diff
@@ -150,39 +150,46 @@ def get_roll_intervals(self, cand_idxs, roll_nom=None, roll_dev=None,
         guides.meta.clear()
         cands = vstack([acqs, guides, self.stars[cols][cand_idxs]])
 
-        q_att = self.att
+        att_targ = self.att_targ
 
         def get_ids_list(roll_offsets):
             ids_list = []
-            # Get the target attitude to roll about from ACA attitude.  For ERs this returns q_att.
-            q_targ = self._calc_targ_from_aca(q_att, y_off, z_off)
 
             for ii, roll_offset in enumerate(roll_offsets):
-                q_targ_roll = Quat([q_targ.ra, q_targ.dec, q_targ.roll + roll_offset])
-                # Transform back to ACA pointing (for ERs this is just q_targ_roll).
-                q_att_roll = self._calc_aca_from_targ(q_targ_roll, y_off, z_off)
+                # Roll about the target attitude, which is offset from ACA attitude by a bit
+                att_targ_rolled = Quat([att_targ.ra, att_targ.dec, att_targ.roll + roll_offset])
+
+                # Transform back to ACA pointing for computing star positions.
+                att_rolled = self._calc_aca_from_targ(att_targ_rolled, y_off, z_off)
 
                 # Get yag/zag row/col for candidates
-                yag, zag = radec_to_yagzag(cands['ra'], cands['dec'], q_att_roll)
+                yag, zag = radec_to_yagzag(cands['ra'], cands['dec'], att_rolled)
                 row, col = yagzag_to_pixels(yag, zag, allow_bad=True, pix_zero_loc='edge')
 
                 ok = (np.abs(row) < CCD['row_max']) & (np.abs(col) < CCD['col_max'])
                 ids_list.append(set(cands['id'][ok]))
             return ids_list
 
+        # If roll_nom and roll_dev not supplied (which is normally the case) compute
+        # them using Sun position.  Here we use the ACA attitude to get pitch since that
+        #  is the official "spacecraft" attitude.
+        att = self.att
         if roll_nom is None or roll_dev is None:
             import Ska.Sun
-            pitch = Ska.Sun.pitch(q_att.ra, q_att.dec, self.date)
+            pitch = Ska.Sun.pitch(att.ra, att.dec, self.date)
         if roll_nom is None:
-            roll_nom = Ska.Sun.nominal_roll(q_att.ra, q_att.dec, self.date)
+            roll_nom = Ska.Sun.nominal_roll(att.ra, att.dec, self.date)
+            att_nom = Quat([att.ra, att.dec, roll_nom])
+            att_nom_targ = self._calc_targ_from_aca(att_nom, y_off, z_off)
+            roll_nom = att_nom_targ.roll
```

Comment:
Another nit:  docstring should probably mention that roll_nom should be supplied in target frame (not that it is really ever supplied).